python/ray/data/datasource/image_folder_datasource.py

```diff
@@ -133,6 +152,97 @@ def _read_file(
         )
 
 
+class _ImageFileMetadataProvider(DefaultFileMetadataProvider):
+    def _set_encoding_ratio(self, encoding_ratio: int):
+        """Set image file encoding ratio, to provide accurate size in bytes metadata."""
+        self._encoding_ratio = encoding_ratio
+
+    def _get_block_metadata(
+        self,
+        paths: List[str],
+        schema: Optional[Union[type, "pyarrow.lib.Schema"]],
+        *,
+        rows_per_file: Optional[int],
+        file_sizes: List[Optional[int]],
+    ) -> BlockMetadata:
+        metadata = super()._get_block_metadata(
+            paths, schema, rows_per_file=rows_per_file, file_sizes=file_sizes
+        )
+        if metadata.size_bytes is not None:
+            metadata.size_bytes = int(metadata.size_bytes * self._encoding_ratio)
+        return metadata
+
+
+class _ImageFolderDatasourceReader(_FileBasedDatasourceReader):
+    def __init__(
+        self,
+        delegate: FileBasedDatasource,
+        paths: List[str],
+        filesystem: "pyarrow.fs.FileSystem",
+        partition_filter: PathPartitionFilter,
+        meta_provider: _ImageFileMetadataProvider = _ImageFileMetadataProvider(),
+        **reader_args,
+    ):
+        super().__init__(
+            delegate=delegate,
+            paths=paths,
+            filesystem=filesystem,
+            schema=None,
+            open_stream_args=None,
+            meta_provider=meta_provider,
+            partition_filter=partition_filter,
+            **reader_args,
+        )
+        self._encoding_ratio = self._estimate_files_encoding_ratio()
+        meta_provider._set_encoding_ratio(self._encoding_ratio)
+
+    def estimate_inmemory_data_size(self) -> Optional[int]:
+        return sum(self._file_sizes) * self._encoding_ratio
+
+    def _estimate_files_encoding_ratio(self) -> float:
```

Comment:
Can you add a bit more documentation in comment about the the estimation method? I think the assumption of estimation method is said well in PR description (just be better to port here in code), i.e. always resizing which make all images same size etc. (I don't know that, it's interesting it's done this way and makes the estimation easier!:)

Reply:
@jianoaix - sure, it's actually always having comment right below at line 204:

```
# Prefetch one image file into memory, and use it to estimate data size for
# all images, because all images are homogeneous with same size after resizing.
```

Wondering what's extra info we need to add here to help provide more clarity? Feel free to suggest a different wording if can be more clearer.

Reply:
@jianoaix - would it be better off with followed?

```
# Prefetch one image file into memory, and use it to estimate data size for
# all images, because all images are homogeneous with same size after resizing.
# Resizing is always enforced when reading every image in ImageFolderDatasource.
```

Reply:
Yes, look good, thanks for making more clear. For me two things seem quite useful to know: we always resize images into same size for two dimensions; and the third dimension will not make image sizes significantly different. These seem the basis we can use this way to estimate (and it will actually work well).

Reply:
@jianoaix - make sense, updated.